packages/common/src/compiler/exploreCompiler.ts

```diff
@@ -62,37 +62,47 @@ export const compileDimensionSql = (
     dimension: Dimension,
     tables: Record<string, Table>,
     quoteChar: string,
-): string => {
+): { sql: string; tablesReferences: Set<string> } => {
     // Dimension might have references to other dimensions
     // Check we don't reference ourself
     const currentRef = `${dimension.table}.${dimension.name}`;
     const currentShortRef = dimension.name;
-    return dimension.sql.replace(lightdashVariablePattern, (_, p1) => {
+    let tablesReferences = new Set([dimension.table]);
+    const sql = dimension.sql.replace(lightdashVariablePattern, (_, p1) => {
         if ([currentShortRef, currentRef].includes(p1)) {
             throw new CompileError(
                 `Dimension "${dimension.name}" in table "${dimension.table}" has a sql string referencing itself: "${dimension.sql}"`,
                 {},
             );
         }
         // eslint-disable-next-line @typescript-eslint/no-use-before-define
-        return compileDimensionReference(
+        const compiledReference = compileDimensionReference(
             p1,
             tables,
             dimension.table,
             quoteChar,
         );
+        tablesReferences = new Set([
+            ...tablesReferences,
+            ...compiledReference.tablesReferences,
+        ]);
+        return compiledReference.sql;
     });
+    return { sql, tablesReferences };
 };
 
 const compileDimensionReference = (
     ref: string,
     tables: Record<string, Table>,
     currentTable: string,
     quoteChar: string,
-): string => {
+): { sql: string; tablesReferences: Set<string> } => {
     // Reference to current table
     if (ref === 'TABLE') {
-        return `${quoteChar}${currentTable}${quoteChar}`;
+        return {
+            sql: `${quoteChar}${currentTable}${quoteChar}`,
+            tablesReferences: new Set([currentTable]),
```

Comment:
While we compile the field sql we now keep track of the tables of the references.